docs/arch/adr-25.md

```diff
@@ -0,0 +1,157 @@
+
+An example of the new sub-task annotation JSON structure:
+
+```json
+{"annotations": [
+  {
+    "task": "T0",
+    "taskType": "drawing",
+    "value": [
+      {
+        "frame": 0,
+        "tool": 0,
+        "toolType": "point",
+        "x": 452.18341064453125,
+        "y": 202.87478637695312,
+        "details": [
+          {"task": "T0.0.0"},
+          {"task": "T0.0.1"}
+        ]
+      },
+      {
+        "frame": 0,
+        "tool": 1,
+        "toolType": "point",
+        "x": 404.61279296875,
+        "y": 583.4398803710938,
+        "value_index": 1,
+        "details": [
+          {"task": "T0.1.0"},
+          {"task": "T0.1.1"}
+        ]
+      }
+    ]
+  },
+  {
+    "task": "T0.0.0",
+    "taskType": "single",
+    "value": 0
+  },
+  {
+    "task": "T0.0.1",
+    "taskType": "dropdown",
+    "value":[
+      {"option-1": 1},
+      {"option-2": 1},
+      {"None": 1}
+    ]
+  },
+  {
+    "task": "T0.1.0",
+    "taskType": "single",
+    "value": 1
+  },
+  {
+    "task": "T0.1.1",
+    "taskType": "dropdown",
+    "value": [
+      {"option-3": 1},
+      {"option-4": 1},
+      {"option-5": 1}
+    ]
+  }
+]}
+```
+
+The sub-task identifiers follow a convention of `TASK_KEY.TOOL_INDEX.DETAILS_INDEX`.
+
+### Drawing sub-task UI
+
+The UI will change to adopt the design of Anti-Slavery Manuscripts (ASM) with this [generalized design](https://projects.invisionapp.com/d/main#/console/12923997/396381420/preview). It will be a pseudo-modal, but with a few notable differences from a true modal:
+
+- The initial position will be near the associated mark made
+- Interactions will be allowed with the image toolbar to allow zoom, rotate, as well as opening of the tutorial, field guide, and task help. Submission of the classification should not be allowed.
+  - If the sub-tasks are required, the modal should not be closeable until the required annotations are made or the mark is deleted if cancelled
+- The dialog can be moved and resized
+
+To support movability and resizing, we will leverage [react-rnd](https://github.com/bokuweb/react-rnd) which is the same library ASM used. Grommet's `Layer` cannot be used since it is intended for actual modal or side panel use and cannot be arbitrarily positioned or moved.
+
+## Status
+
+Proposed
+
+## Consequences
+
+Flattening the annotations array for drawing task sub-tasks is conceptually consistent with the move to using workflow steps to flatten the combo task. It is however a breaking change and this change will have to be communicated to project builders. As with other classifications from the new classifier, this can be checked by the presence of `classifier_version: 2.0` in the classification metadata. In the future, we would also like to include a link to json schema for each annotation type as recommended in [ADR 07](adr-07.md).
```

Comment:
Will this change involve changes to the classification API eg. can we still use `classification.addAnnotation(task, value)` from our task components?

Ideally, I think I'd like a model that extends the existing classifier model without breaking existing tasks like the PH-TESS data task.

Reply:
I don't think so. We just need to make sure the `task` identifier matches up to what is in the `details` array. I think it would involve calling `classification.addAnnotation` twice. Once for the drawing task to update its annotation for `details` and once for the mark's sub-task annotation. I think the flattening has benefits for raw classification exports too, not just aggregation, but I could be wrong. Perhaps researcher input would benefit here too, so I'll invite the same as we invited on #1390.

Reply:
If we use array index as a mark identifier while classifying then this won't work since the mark array index changes when we add and delete marks. I was under the impression that the structure proposed here isn't written to the classification until the `completeClassification` action fires.

Reply:
In fact, marks is a map, not an array, so this definitely needs more thought.

Reply:
On example, of the top of my head: when I click on a mark to edit the sub tasks, how are the subtask annotations retrieved from the store?

I don't think the model proposed here works while classification is in progress.

Reply:
This was discussed in another comment thread: https://github.com/zooniverse/front-end-monorepo/pull/1397#discussion_r364835591